gateway/src/apicast/policy/upstream/upstream.lua

```diff
@@ -49,7 +49,8 @@ function _M:rewrite(context)
   for _, rule in ipairs(self.rules) do
     if match(req_uri, rule.regex) then
       ngx.log(ngx.DEBUG, 'upstream policy uri: ', req_uri, ' regex: ', rule.regex, ' match: true')
-      context[self] = rule.upstream
+      -- better to allocate new object for each request as it is going to get mutated
+      context[self] = Upstream.new(rule.url)
```

Comment:
In this case, it doesn't matter much because we're just storing one object, but it might be a better practice to namespace what we store in the context: `context[self].url = ... `

Reply:
Another option would be to store just the url instead of the Upstream instance.
The Upstream can be instantiated in `content()`. Not a big difference, though.

Reply:
We will have to try to reduce the allocations in the future, so I'm trying to not create objects when it is not absolutely necessary. In this case doing a namespace would require another table.

I'm parsing the upstream here, so potential failure will show early in the rewrite phase before other calls are made (like auth).